### ISSUE TITLE
Show/hide Password button label is not meaningful enough and it's redundantly announced

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4640]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 2.4.6 Headings and Labels

**Test Environment:**
- Test URL: https://clarity.design/documentation/login
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 2020.1

**Steps to reproduce:**
1. Open the test URL
2. Move screen reader focus on show/hide password button directly or out of context and check if the label announced appropriately.

**Expected result:**
When screen reader user move focus on the show/hide password button by using keyboard shortcuts like button shortcut key "B", user should be able to understand the purpose of that button clearly.

**Actual result:**
In this case user just hear "show", which doesn't convey the meaning clearly if the button is focused directly using keyboard shortcut.
Also, the label is announced twice as the label and title are same.

**Comment by gkhadK** [comment]
Same issue is observed on the password component page.
